lib/elements/svg_objects.py

```diff
@@ -0,0 +1,198 @@
+from cubicsuperpath import parsePath
+from simpletransform import applyTransformToPoint, roughBBox
+
+from ..i18n import _
+from ..svg import get_node_transform
+from ..svg.svg import find_elements
+from ..svg.tags import (EMBROIDERABLE_TAGS, SVG_CIRCLE_TAG, SVG_ELLIPSE_TAG,
+                        SVG_IMAGE_TAG, SVG_RECT_TAG, SVG_TEXT_TAG)
+from .validation import ValidationTypeWarning
+
+
+class TextTypeWarning(ValidationTypeWarning):
+    name = _("Text")
+    description = _("Ink/Stitch cannot work with objects like text.")
+    steps_to_solve = [
+        _('* Text: Create your own letters or try the lettering tool:'),
+        _('- Extensions > Ink/Stitch > Lettering')
+    ]
+
+
+class ImageTypeWarning(ValidationTypeWarning):
+    name = _("Image")
+    description = _("Ink/Stitch can't work with objects like images.")
+    steps_to_solve = [
+        _('* Convert your image into a path: Path > Trace Bitmap... (Shift+Alt+B) '
+          '(further steps might be required)'),
+        _('* Alternatively redraw the image with the pen (P) or bezier (B) tool')
+    ]
+
+
+class NoColorWarning(ValidationTypeWarning):
+    name = _("No color settings")
+    description = _("There is no color information for this object. "
+                    "Ink/Stitch doesn't know, wether it was meant to be a stroke or a fill object. "
+                    "It will be rendered as fill, until a stroke color is set.")
+    steps_to_solve = [
+        _('Go to: Objects > Fill and Stroke ...'),
+        _('Apply a fill or a stroke color to the object.')
+    ]
+
+
+class SVGObjects(object):
```

Comment:
I'm wondering if we should just create full EmbroideryElement subclasses for each of these.  We can add warnings using the normal `validation_warnings()` method.  For the `to_patches()` method, we can just return an empty list which will effectively ignore the object (just like it is now).

That also prevents us from having to do all that xpath stuff below.  We'd just handle them in the `nodes_to_elements()` function just like normal paths.

What do you think?

Reply:
Hm, interesting idea. I will look into it. But I might not find the time for it today or tomorrow.